src/main/java/org/folio/service/pieces/PiecesClaimingService.java

```diff
@@ -0,0 +1,238 @@
+      }
+    });
+    return pieceIdByVendorIdFutures;
+  }
+
+  private Future<Pair<String, String>> createVendorPiecePair(Pair<String, String> piecePoLinePairs,
+                                                             Piece piece, RequestContext requestContext) {
+    if (!piece.getReceivingStatus().equals(Piece.ReceivingStatus.LATE)) {
+      log.info("createVendorPiecePair:: Ignoring processing of a piece not in LATE state, piece id: {}", piece.getId());
+      return Future.succeededFuture();
+    }
+    return purchaseOrderLineService.getOrderLineById(piecePoLinePairs.getLeft(), requestContext)
+      .compose(poLine -> purchaseOrderStorageService.getPurchaseOrderById(poLine.getPurchaseOrderId(), requestContext)
+      .compose(purchaseOrder -> organizationService.getVendorById(purchaseOrder.getVendor(), requestContext)))
+      .map(vendor -> {
+        if (Objects.nonNull(vendor) && Boolean.TRUE.equals(vendor.getIsVendor())) {
+          return Pair.of(vendor.getId(), piecePoLinePairs.getRight());
+        }
+        return null;
+      });
+  }
+
+  private static Map<String, List<String>> transformAndGroupPieceIdsByVendorId(List<Pair<String, String>> piecesByVendorList) {
+    return StreamEx.of(piecesByVendorList).distinct().filter(Objects::nonNull)
+      .groupingBy(Pair::getKey, mapping(Pair::getValue, toList()));
+  }
+
+  private Future<ClaimingResults> createJobsByVendor(JsonObject config, Map<String, List<String>> pieceIdsByVendorId,
+                                                     RequestContext requestContext) {
+    log.info("createJobsByVendor:: Creating jobs by vendor, vendors by pieces count: {}", pieceIdsByVendorId.size());
+    if (CollectionUtils.isEmpty(pieceIdsByVendorId)) {
+      log.info("createJobsByVendor:: No jobs are created, pieceIdsByVendorId is empty");
+      return Future.succeededFuture(new ClaimingResults()
+        .withClaimingPieceResults(createErrorClaimingResults(pieceIdsByVendorId, CANNOT_GROUP_PIECES_BY_VENDOR_MESSAGE.getValue())));
+    }
+    return collectResultsOnSuccess(createUpdatePiecesAndJobFutures(config, pieceIdsByVendorId, requestContext))
+      .map(updatedPieceLists -> {
+        if (CollectionUtils.isEmpty(updatedPieceLists)) {
+          log.info("createJobsByVendor:: No pieces were processed for claiming");
+          return new ClaimingResults().withClaimingPieceResults(createErrorClaimingResults(pieceIdsByVendorId, CANNOT_CREATE_JOBS_AND_UPDATE_PIECES.getValue()));
+        }
+        var successClaimingPieceResults = createSuccessClaimingResults(updatedPieceLists);
+        log.info("createJobsByVendor:: Successfully processed pieces for claiming, count: {}", successClaimingPieceResults.size());
+        var claimingResults = new ClaimingResults().withClaimingPieceResults(successClaimingPieceResults);
+        log.info("createJobsByVendor:: Returning claiming results, claimingResults: {}", JsonObject.mapFrom(claimingResults).encodePrettily());
```

Comment:
I think we can move to debug, since JsonObject.mapFrom(claimingResults).encodePrettily() is being used

Reply:
I prefer to have this as is at the moment, plus it has no sensitive data

Reply:
> I think we can move to debug, since JsonObject.mapFrom(claimingResults).encodePrettily() is being used

Okay will move to debug